xbmc/pvr/windows/GUIWindowPVRBase.cpp

```diff
@@ -587,8 +587,12 @@ bool CGUIWindowPVRBase::EditTimer(CFileItem *item)
     return false;
   }
 
-  if (ShowTimerSettings(timer) && !timer->GetTimerType()->IsReadOnly())
-    return g_PVRTimers->UpdateTimer(timer);
+  // clone the timer.
+  const CPVRTimerInfoTagPtr newTimer(new CPVRTimerInfoTag);
+  newTimer->UpdateEntry(timer);
+
+  if (ShowTimerSettings(newTimer) && !timer->GetTimerType()->IsReadOnly())
+    return g_PVRTimers->UpdateTimer(newTimer);
```

Comment:
> Does this mean timer updates will only be reflected in the UI once the backend has processes the request and triggered Kodi to update timers?

Exactly. My take is that this was always the intention of the original authors of the code, but it was not correctly implemented or it broke at some point in time (maybe even by changes I did).